File: ios/Classes/FlutterDrivingDirectionsPlugin.m
#import "FlutterDrivingDirectionsPlugin.h"

#import <MapKit/MapKit.h>

@implementation FlutterDrivingDirectionsPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_driving_directions"
            binaryMessenger:[registrar messenger]];
  FlutterDrivingDirectionsPlugin* instance = [[FlutterDrivingDirectionsPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {

  if ([@"launchDirections" isEqualToString:call.method]) {
    result(FlutterMethodNotImplemented);
  } else {
    NSNumber *latitude = call.arguments[@"latitude"];
    NSNumber *longitude = call.arguments[@"longitude"];
    NSString *address = call.arguments[@"address"];

    if (@available(iOS 10.0, *)) {
        CLLocationCoordinate2D coordinate = CLLocationCoordinate2DMake([latitude doubleValue], [longitude doubleValue]);
        MKPlacemark *placemark = [[MKPlacemark alloc] initWithCoordinate: coordinate];
        MKMapItem *mapItem = [[MKMapItem alloc] initWithPlacemark: placemark];
        mapItem.name = address;
        NSDictionary *options = [[NSDictionary alloc] initWithObjectsAndKeys:
            MKLaunchOptionsDirectionsModeDriving, MKLaunchOptionsDirectionsModeKey, nil];
        [mapItem openInMapsWithLaunchOptions: options];
    }
  }
}

@end

File: ios/flutter_driving_directions/Sources/flutter_driving_directions/FlutterDrivingDirectionsPlugin.swift
import Flutter
import UIKit
import MapKit

public class FlutterDrivingDirectionsPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_driving_directions", binaryMessenger: registrar.messenger())
    let instance = FlutterDrivingDirectionsPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    let arguments = call.arguments as! Dictionary<String, Any>
    let latitude = arguments["latitude"] as! Double
    let longitude = arguments["longitude"] as! Double
    let label = arguments["label"] as! String

    let coordinate = CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
    let placemark = MKPlacemark(coordinate: coordinate)
    let mapItem = MKMapItem(placemark: placemark)
    mapItem.name = label
    mapItem.openInMaps(launchOptions: [MKLaunchOptionsDirectionsModeKey: MKLaunchOptionsDirectionsModeDriving])
  }
}
